sshd-common/src/main/java/org/apache/sshd/common/config/keys/loader/openssh/OpenSSHKeyPairResourceParser.java

```diff
@@ -182,9 +242,8 @@ protected PublicKey readPublicKey(
         boolean traceEnabled = log.isTraceEnabled();
         int check1 = KeyEntryResolver.decodeInt(stream);
         int check2 = KeyEntryResolver.decodeInt(stream);
-        if (traceEnabled) {
-            log.trace("readPrivateKeys({}) check1=0x{}, check2=0x{}",
-                resourceKey, Integer.toHexString(check1), Integer.toHexString(check2));
+        if (check1 != check2) {
+            throw new StreamCorruptedException(format("Check number mismatch in key {0}", resourceKey.getName()));
         }
```

Comment:
Please don't use `MessageTextFormat#format` - there is no upside in this case...